functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -189,7 +241,14 @@ Then it will export the results to Export-DbaDiagnosticQuery.
             }
 
             if (!$instanceOnly) {
-                $databases = $server.Query("Select Name from sys.databases where name not in ('master', 'model', 'msdb', 'tempdb')")
+                if (-not $DatabaseName) {
+                    $databases = Get-DbaDatabase -SqlInstance $SqlInstance -ExcludeAllSystemDb  | select-object Name
+                    Write-Message -Level Debug -Message  "No Database Name Filtered: Pulled $(@($Databases).count)" -Verbose
+                }
+                else {
+                    $databases = Get-DbaDatabase -SqlInstance $SqlInstance -ExcludeAllSystemDb -Database $DatabaseName | select-object Name
+                    Write-Message -Level Debug -Message  "Filtered to databases: $(($Databases | select-object -ExpandProperty Name) -join ',')" -Verbose
+                }
             }
 
             $parsedscript = $scriptversions | Where-Object -Property Version -eq $version | Select-Object -ExpandProperty Script
```

Comment:
if you only need Script, use `($object).property` notation instead of -ExpandProperty

Reply:
Since I in the past used strict-mode, using the syntax you mention above fails, while ExpandProperty handles property not existing without issues. 
Can you educate me on 2 things: 
1- wouldn't it be @($Object).property to ensure it recognizes it's an object?
2- Is it considered ok to use .property, even though property doesn't exist? I'm willing to change my approach overall, just want to understand why a little better.

Reply:
sorry, typo. use `$object.property`.

forget strict-mode. dbatools doesn't run with strict mode.

straight from https://github.com/sqlcollaborative/dbatools/wiki/Do-And-Do-Not . 
> Prefer dot notation over Select -ExpandProperty.


as for point 2)
> Is it considered ok to use .property, even though property doesn't exist? I'm willing to change my approach overall, just want to understand why a little better.

```
PS > $x.b | gm
gm : You must specify an object for the Get-Member cmdlet.
At line:1 char:8
+ $x.b | gm
+        ~~
    + CategoryInfo          : CloseError: (:) [Get-Member], InvalidOperationException
    + FullyQualifiedErrorId : NoObjectInGetMember,Microsoft.PowerShell.Commands.GetMemberCommand

PS > ($x | select-object -ExpandProperty b) | gm
select-object : Property "b" cannot be found.
At line:1 char:7
+ ($x | select-object -ExpandProperty b) | gm
+       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (@{a=1}:PSObject) [Select-Object], PSArgumentException
    + FullyQualifiedErrorId : ExpandPropertyNotFound,Microsoft.PowerShell.Commands.SelectObjectCommand

gm : You must specify an object for the Get-Member cmdlet.
At line:1 char:42
+ ($x | select-object -ExpandProperty b) | gm
+                                          ~~
    + CategoryInfo          : CloseError: (:) [Get-Member], InvalidOperationException
    + FullyQualifiedErrorId : NoObjectInGetMember,Microsoft.PowerShell.Commands.GetMemberCommand
```
both will end in null. -expandproperty will raise ..... but no code here is actually checking (the exception or the being null). 
So go for the dot notation which is:
- shorter 
- easier to read 
- for arrays of objects is a ton faster than passing down the dreaded `|`
- doesn't raise, so you don't need additional code for try/catch
- in the even you need to assure result is there, just check for $null

Reply:
This is great. Thanks for this. 
I'm assuming that for most dev work you find not using strict-mode and this type of approach is a much easier/cleaner to work with?

Thanks again for the help and resources, will use to improve my next pulls.